kroxylicious-schema-tools/kroxylicious-schema-tool/src/main/java/io/kroxylicious/tools/schema/compiler/SchemaCompiler.java

```diff
@@ -0,0 +1,233 @@
+                        return parseSchema(srcPath, file);
+                    }
+                    return Stream.empty();
+                }).toList().stream(); // Need to materialise this list else the try(walkPaths) will close Stream before anything pulled through the stream
+            }
+            catch (IOException e) {
+                throw new UncheckedIOException("Unable to walk source directory " + srcPath, e);
+            }
+        }).toList();
+    }
+
+    private Stream<SchemaInput> parseSchema(Path srcPath, Path schemaFile) {
+        try {
+            var relPath = srcPath.relativize(schemaFile);
+            LOGGER.debug("Parsing {}", schemaFile);
+            var tree = mapper.readTree(schemaFile.toFile());
+
+            JsonNode schemaKeywordNode = tree.path("$schema");
+            if (schemaKeywordNode.isMissingNode()) {
+                diagnostics.reportWarning("Ignoring non-schema file: {}", schemaFile);
+                return Stream.empty();
+            }
+            if (!"http://json-schema.org/draft-04/schema#".equals(schemaKeywordNode.asText("http://json-schema.org/draft-04/schema#"))) {
+                diagnostics.reportWarning("Ignoring non-schema file: {}", schemaFile);
+                return Stream.empty();
+            }
+            var rootSchema = mapper.convertValue(tree, SchemaObject.class);
+
+            // Build the map of absolute URI identifiers to schema
+            rootSchema.visitSchemas(schemaFile.toUri(), idVisitor);
+
+            // We should now be able to resolve local $ref
+            String rootClass = schemaFile.getFileName().toString().replaceAll("\\.yaml$", "");
+            var typeNameVisitor = new TypeNameVisitor(diagnostics, idVisitor, rootClass);
+            rootSchema.visitSchemas(schemaFile.toUri(), typeNameVisitor);
+
+            String pkg = StreamSupport.stream(relPath.getParent().spliterator(), false)
+                    .map(Path::toString)
+                    .collect(Collectors.joining("."));
+
+            return Stream.of(new SchemaInput(schemaFile, pkg, rootSchema));
+        }
+        catch (IOException | IllegalArgumentException | VisitException e) {
+            diagnostics.reportError("Unable to read source file {}: {}", schemaFile, e.getMessage());
+            return Stream.empty();
+        }
+    }
+
+    public Stream<CompilationUnit> gen(List<SchemaInput> inputs) {
+
+        return inputs.stream()
+                .flatMap(input -> {
+                    try {
+                        return codeGen.genDecls(input).stream();
+                    }
+                    catch (VisitException e) {
+                        diagnostics.reportFatal("Error: {}", e.getMessage(), e);
+                        return Stream.empty();
+                    }
+                });
+
+    }
+
+    public int numFatals() {
+        return diagnostics.getNumFatals();
```

Comment:
This should only ever be `0` or `1` right? At least thats what I expect from the javadoc on [FatalException](https://github.com/kroxylicious/kroxylicious/pull/1639/files#diff-148beca53eb58754f84a45b20e18a74a03a15c774d494d7a1810891883081d7fR10)